include/kframework/lemmas/lemmas.k

```diff
@@ -147,13 +146,13 @@ module LEMMAS [symbolic]
 
     rule #padToWidth(32, #asByteStack(V)) => #buf(32, V)  requires #rangeUInt(256, V) [simplification]
 
-    rule #asWord(WS) >>Int M => #asWord(WS [ 0 .. #sizeByteArray(WS) -Int (M /Int 8) ])  requires 0 <=Int M andBool M modInt 8 ==Int 0 [simplification]
+    rule #asWord(WS) >>Int M => #asWord( #range(WS, 0, #sizeByteArray(WS) -Int (M /Int 8) ))  requires 0 <=Int M andBool M modInt 8 ==Int 0 [simplification]
 
     rule #asWord(#padRightToWidth(32, BUF)) &Int notMaxUInt224 =>
          #asWord(#padRightToWidth(32, BUF))
       requires #sizeByteArray(BUF) <=Int 4 [simplification]
 
-    rule #padRightToWidth(_, BUF)[ 0 .. WIDTH ] => BUF requires #sizeByteArray(BUF) ==Int WIDTH [simplification]
+    rule #range(#padRightToWidth(_, BUF), 0, WIDTH) => BUF requires #sizeByteArray(BUF) ==Int WIDTH [simplification]
 
```

Comment:
I would try just removing simplification rules like this instead of adapting them.

Reply:
Done.